zerver/tests/test_message_send.py

```diff
@@ -1321,6 +1321,68 @@ def test_message_bot_mentions(self) -> None:
         self.assertEqual(user_message.message.content, content)
         self.assertTrue(user_message.flags.mentioned)
 
+    def test_wildcard_mention_restrictions(self) -> None:
+        cordelia = self.example_user('cordelia')
+        iago = self.example_user('iago')
+        self.login_user(cordelia)
+
+        stream_name = 'test_stream'
+        self.subscribe(cordelia, stream_name)
+        self.subscribe(iago, stream_name)
+
+        # When wildcard_mention_policy is set to WILDCARD_MENTION_SHOW_WARNING and a message with wildcard mention
+        # is sent through API, we simply send the message. Warning is only shown when message is sent through UI.
+        content = '@**all** test wildcard mention'
+        result = self.client_post("/json/messages", {"type": "stream",
+                                                     "to": "test_stream",
+                                                     "client": "test suite",
+                                                     "content": content,
+                                                     "topic": "Test topic"})
+        self.assert_json_success(result)
+        last_message = self.get_last_message()
+        self.assertEqual(last_message.content, content)
+
+        do_set_realm_property(cordelia.realm, 'wildcard_mention_policy', Realm.WILDCARD_MENTION_ALLOW_EVERYONE)
+        result = self.client_post("/json/messages", {"type": "stream",
```

Comment:
It might be worth running this new chunk of code through Black, as we're probably eventually gonna go away from the hanging-indent style.

Reply:
But I am not sure how to run this code through Black. Should it be done from commits from #15662?

Reply:
Install black (google “Python black code formatter” or similar).  Copy the big chunk of code here to temp.py and run black against it.  Then paste back into the existing file.  Make sense?

Reply:
Yeah got it.

Reply:
Updated the code with Black formatting.